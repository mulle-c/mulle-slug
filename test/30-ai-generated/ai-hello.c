#include <stdio.h>

#include <mulle-buffer/mulle-buffer.h>
#include <mulle-utf/mulle-utf.h>
#include <mulle-slug/mulle-slug.h>


void test_mulle_buffer_slugify_utf8data(const char *input) {
    struct mulle_buffer buffer;
    mulle_buffer_init(&buffer, NULL);
    
    struct mulle_utf8data data;
    data.characters = (char *)input;
    data.length = strlen(input);
    
    mulle_buffer_slugify_utf8data(&buffer, data);
    
    printf("Input: %s\n", input);
    printf("Output: %s\n\n", buffer._storage);
    
    mulle_buffer_done(&buffer);
}

int main() {
    test_mulle_buffer_slugify_utf8data("Hello, World!");
    test_mulle_buffer_slugify_utf8data(""); // Empty string
    test_mulle_buffer_slugify_utf8data("üôÇ Emoji test üëç"); // Emoji content
    test_mulle_buffer_slugify_utf8data("Test!#$%&*()_+-={}[]|;:'\"<>,.?/"); // Special characters
    test_mulle_buffer_slugify_utf8data("–ü—Ä–∏–≤–µ—Ç –º–∏—Ä!"); // Non-English characters (Cyrillic)

    return 0;
}
