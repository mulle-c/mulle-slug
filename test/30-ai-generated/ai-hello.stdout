Input: Hello, World!
Output: Hello-World

Input: 
Output: 

Input: 🙂 Emoji test 👍
Output: Emoji-test

Input: Test!#$%&*()_+-={}[]|;:'"<>,.?/
Output: Test-hashdollar-lessgreater

Input: Привет мир!
Output: Privet-mir

